src/bp/nlu-core/entities/microsoft-extractor/index.ts

```diff
@@ -0,0 +1,380 @@
+  private _cache: SystemEntityCacheManager
+
+  constructor(private readonly logger?: NLU.Logger) {
+    this.logger = logger
+    this._cache = new SystemEntityCacheManager(
+      path.join(process.APP_DATA_PATH || '', 'cache', 'microsoft_sys_entities.json'),
+      true,
+      logger
+    )
+  }
+
+  private _langToCulture = (lang: SupportedLangs): string => {
+    switch (lang) {
+      case 'zh':
+        return Recognizers.Culture.Chinese
+      case 'nl':
+        return Recognizers.Culture.Dutch
+      case 'en':
+        return Recognizers.Culture.English
+      case 'fr':
+        return Recognizers.Culture.French
+      case 'de':
+        return Recognizers.Culture.German
+      case 'it':
+        return Recognizers.Culture.Italian
+      case 'ja':
+        return Recognizers.Culture.Japanese
+      case 'pt':
+        return Recognizers.Culture.Portuguese
+      case 'es':
+        return Recognizers.Culture.Spanish
+      default:
+        throw new Error('This language is not supported by Microsoft Recognizer')
+    }
+  }
+
+  private _globalRecognizers = [
+    Recognizers.recognizePhoneNumber,
+    Recognizers.recognizeIpAddress,
+    Recognizers.recognizeMention,
+    Recognizers.recognizeHashtag,
+    Recognizers.recognizeEmail,
+    Recognizers.recognizeURL,
+    Recognizers.recognizeGUID
+  ]
+
+  private _languageDependantRecognizers = [
+    Recognizers.recognizeNumber,
+    Recognizers.recognizeOrdinal,
+    Recognizers.recognizePercentage,
+    Recognizers.recognizeAge,
+    Recognizers.recognizeCurrency,
+    Recognizers.recognizeDimension,
+    Recognizers.recognizeTemperature,
+    Recognizers.recognizeDateTime,
+    Recognizers.recognizeBoolean
+  ]
+
+  public static get entityTypes(): string[] {
+    return []
+  }
+
+  public async configure() {
+    await this._cache.restoreCache()
+  }
+
+  public async extractMultiple(
+    inputs: string[],
+    lang: string,
+    useCache?: boolean
+  ): Promise<EntityExtractionResult[][]> {
+    let options: MicrosoftParams
+
+    if (!SupportedLangsList.includes(lang)) {
+      lang = 'en'
+      options = {
+        lang: 'en' as SupportedLangs,
+        recognizers: [this._globalRecognizers]
+      }
+    }
+
+    options = {
+      lang: lang as SupportedLangs,
+      recognizers: [...this._languageDependantRecognizers, ...this._globalRecognizers]
+    }
+
+    const [cached, toFetch] = this._cache.getCachedAndToFetch(inputs)
+
+    const chunks = _.chunk(toFetch, BATCH_SIZE)
+    const batchedRes = await Promise.mapSeries(chunks, c => this._extractBatch(c, options))
+
+    return _.chain(batchedRes)
+      .flatten()
+      .concat(cached)
+      .orderBy('idx')
+      .map(x => x.entities!)
+      .value()
+  }
+
+  public async extract(input: string, lang: string, useCache?: boolean): Promise<EntityExtractionResult[]> {
+    return (await this.extractMultiple([input], lang, useCache))[0]
+  }
+
+  private formatEntity(entity) {
```

Comment:
Don't leave empty typings even in private functions. This has the following benefits:

1 - It really helps another developper review your code 
2 - If we ever change the definition of type `Entity` in a file 1000000km away, the compiler will break right here and tell you that you don't respect correct typings.

Reply:
I just don't know every possibilites for the microsoft entity... I added a typing but hoping I though of all the cases